blueflood-core/src/main/java/com/rackspacecloud/blueflood/utils/GenericClassLoader.java

```diff
@@ -0,0 +1,74 @@
+
+/*
+ * Copyright 2015 Rackspace
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.rackspacecloud.blueflood.utils;
+
+import com.rackspacecloud.blueflood.service.Configuration;
+import com.rackspacecloud.blueflood.service.CoreConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+
+
+public class GenericClassLoader {
+    private static final Logger log = LoggerFactory.getLogger(GenericClassLoader.class);
+
+    private static Map<CoreConfig, Object> loadedModules = new HashMap<CoreConfig, Object>();
+
+    public static Object getGenericInstance(Class c, CoreConfig moduleName){
+
+        Object genericInstance = loadedModules.get(moduleName);
+        if (genericInstance != null)
+            return genericInstance;
+
```

Comment:
This can have severe side effects. If the cached object is of type EventsElasticIO and is cast as DiscoveryIO, it will crash. I think we are not going to have the singleton functionality in the module loader. Every time someone invokes getInstance they will get a new instance. The responsibility of caching the instance as a static field should now lie with the invoking classes.